### PR TITLE
Span builder MVP: in-memory

### PR DESCRIPTION
**Pull request by goorui**

### CODE REVIEW

File: span-builder/constants.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+GET_PROVIDERS_CLIENT = "GET_PROVIDERS_CLIENT"
+GET_PROVIDERS_SERVER = "GET_PROVIDERS_SERVER" 
+BITSWAP_CLIENT = "BITSWAP_CLIENT"
+BITSWAP_SERVER = "BITSWAP_SERVER"
+READ_FROM_FILE_STORE = "READ_FROM_FILE_STORE"
+
+NABU_EVENT_TYPES = [
+	GET_PROVIDERS_CLIENT,
+	GET_PROVIDERS_SERVER,
+    BITSWAP_CLIENT,
```
Comment (sonudoo) [inline span-builder/constants.py:51]: nit: Indent issue.

File: span-builder/service.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+def construct_span_id_from_span(trace_id, node_id, thread_id, span_name):
+    span_id = f"{trace_id}_{node_id}_{thread_id}_{span_name}"
+    span_id = hashlib.md5(span_id.encode('utf-8')).hexdigest()[:16]
+    return span_id
+
+
+def build_parent_child_spans(trace_id: str):
+    spans = []
+
+    def find_span(node_id: str, type: str):
```
Comment (sonudoo) [inline span-builder/service.py:68]: Can this method instead return a reference to the span? Similarly, return None if not found.

File: span-builder/service.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        self.children = children if children is not None else []
+
+
+def construct_span_id_from_span(trace_id, node_id, thread_id, span_name):
+    span_id = f"{trace_id}_{node_id}_{thread_id}_{span_name}"
+    span_id = hashlib.md5(span_id.encode('utf-8')).hexdigest()[:16]
+    return span_id
+
+
+def build_parent_child_spans(trace_id: str):
```
Comment (sonudoo) [inline span-builder/service.py:65]: There is a way to unify the logic. See Figure 1 in the report doc.

File: span-builder/service.py
Comment (sonudoo) [inline span-builder/service.py:1]: Can you rearrange the order of functions? May be move all API (app.route) methods above the non-API ones.

**Review by sonudoo — APPROVED** [review]

**Review by millerm — APPROVED** [review]

**Pull request merged**